src/data_reader/buf_reader.rs

```diff
@@ -0,0 +1,66 @@
+// Pete Walsh's code for a buffered read over compressed files.
+
+use std::{
+    fs::File,
+    io::{self, prelude::*},
+    rc::Rc,
+};
+
+use anyhow::Result;
+use flate2::read::MultiGzDecoder;
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct DataInstance {
```

Comment:
Doesn't look like we need this struct? We used this throughout WIMBD because we didn't care about other metadata with the JSON blobs.